apps/ledger-live-desktop/src/renderer/components/SignMessageConfirm/index.tsx

```diff
@@ -12,29 +12,50 @@ import useTheme from "~/renderer/hooks/useTheme";
 import Text from "~/renderer/components/Text";
 import Box from "~/renderer/components/Box";
 import { getLLDCoinFamily } from "~/renderer/families";
+import FormattedVal from "~/renderer/components/FormattedVal";
 
 const FieldText = styled(Text).attrs(() => ({
   ml: 1,
   ff: "Inter|Medium",
   color: "palette.text.shade80",
   fontSize: 3,
-}))`
-  word-break: break-all;
-  text-align: right;
-  max-width: 50%;
-`;
+}))``;
 
 export type FieldComponentProps = {
-  account: AccountLike;
+  account: Account;
   field: DeviceTransactionField;
+  contractAddress?: string;
 };
 
 export type FieldComponent = React.ComponentType<FieldComponentProps>;
 
-const TextField = ({ field }: FieldComponentProps) => {
+const TextField = ({ field, account, contractAddress }: FieldComponentProps) => {
+  const isPropertyAmount = field.label.includes("Amount") && contractAddress;
+
+  const unit = isPropertyAmount
+    ? account.subAccounts?.find(a => a.token.contractAddress === contractAddress)?.token.units[0]
+    : undefined;
+
   return field.type === "text" ? (
     <SignMessageConfirmField label={field.label}>
-      <FieldText>{field.value}</FieldText>
+      <FieldText>
+        {isPropertyAmount ? (
```

Comment:
should make a check on `unit` instead, we could end up in this situation, as `<FormattedVal>` has an invariant that checks for `unit`: 

<img width="744" alt="image" src="https://github.com/user-attachments/assets/53f766f9-d0ca-450f-a1ab-a5345e245c93" />
```suggestion
        {unit ? (
```

Reply:
I've fixed it 6aae17f679200545860862bb70fde2421677caee

Reply:
@Canestin Why did we resolve this while it's still using the `isPropertyAmount` ?

Edit: looks like it was only updated for this file `apps/ledger-live-desktop/src/renderer/modals/SignMessage/steps/StepSummary.tsx`

Reply:
https://github.com/LedgerHQ/ledger-live/pull/9158/commits/1d9344cac302be9d4dbe3ec3c9d0a804569684d5#diff-15a820817a850003c96032d59991c4b0648434571bf22737f00d331d1842c5f0L33-R41